spras/evaluation.py

```diff
@@ -380,13 +378,40 @@ def precision_recall_curve_node_ensemble(node_ensembles: dict, node_table: pd.Da
 
         prc_dfs = []
         metric_dfs = []
-
+        prc_input_nodes_baseline_df = None
         baseline = None
 
         for label, node_ensemble in node_ensembles.items():
             if not node_ensemble.empty:
                 y_true = [1 if node in gold_standard_nodes else 0 for node in node_ensemble['Node']]
                 y_scores = node_ensemble['Frequency'].tolist()
+
+                # input nodes (sources, targets, prizes, actives) may be easier to recover but are still valid gold standard nodes;
+                # the Input_Nodes_Baseline PR curve highlights their overlap with the gold standard.
+                if prc_input_nodes_baseline_df is None:
+                    input_nodes_set = set(input_nodes['NODEID'])
+                    input_nodes_gold_intersection = input_nodes_set & gold_standard_nodes # TODO should this be all inputs nodes or the intersection with the gold standard for this baseline? I think it should be the intersection
+                    input_nodes_ensemble_df = node_ensemble.copy()
+
+                    input_nodes_ensemble_df["Frequency"] = (
```

Comment:
It would help to have a comment explaining what this is doing. I understand it. It's for future reference.